app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,67 @@
+class OffersController < ApplicationController
+  before_action :set_offer, only: [:show, :edit, :update, :destroy]
+  before_action :require_offer_owner!, only: [:edit, :update, :destroy]
+
+  def index
+    @q = Offer.where("state = ?", true).ransack(params[:q])
```

Comment:
What is `state = true` supposed to represent in your database?
This can also be rewritten as to have a less string based api
```suggestion
    @q = Offer.where(state: true).ransack(params[:q])
```